Services/source/housekeeping/housekeeping_to_beacon.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (C) 2015  University of Alberta
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+/**
+ * @file housekeeping_to_beacon.c
+ * @author Grace Yi
+ * @date 2021-10-27
+ */
+#include "housekeeping_to_beacon.h"
+#include "housekeeping_service.h" //contains all housekeeping data
+#include "rtcmk.h" //to get time from RTC
+#include "services.h"
+#include "ex2_hal/ex2_adcs_software/equipment_handler/inc/adcs_handler.h"
+#include "ex2_system/include/beacon/beacon_task.h" //contains beacon packets (ie. a summary of housekeeping data)
+
+/**
+ * @brief
+ *      Updates the beacon packet with the latest housekeeping data
+ * @param all_hk_data
+ *      The pointer to where housekeeping data is kept
+ */
+
+beacon_t beacon_packet;
+int i;
+
+/* Updates Beacon Packet with the latest housekeeping data */
+void update_beacon(All_systems_housekeeping* all_hk_data) {
+  //Populate the beacon packet by fetching relevant data from housekeeping_services.c
+  //get the unix time from RTC, and convert it to a struct using RTCMK_GetUnix
+  //RTCMK_GetUnix(&(beacon_packet.time));
+  memcpy(&(beacon_packet.time), &(all_hk_data->hk_timeorder.UNIXtimestamp), sizeof(beacon_packet.time));
+ 
+  /*-------EPS-------*/
+  memcpy(&(beacon_packet.eps_mode), &(all_hk_data->EPS_hk.battMode), sizeof(beacon_packet.eps_mode));
+  memcpy(&(beacon_packet.battery_voltage), &(all_hk_data->EPS_hk.vBatt), sizeof(beacon_packet.battery_voltage));
+  memcpy(&(beacon_packet.battery_input_current), &(all_hk_data->EPS_hk.curBattIn), sizeof(beacon_packet.battery_input_current));
+  for (i = 0; i < 10; i++) {
+    memcpy(&(beacon_packet.current_channels[i]), &(all_hk_data->EPS_hk.curOutput[i]), sizeof(uint16_t));
+  }  
+  memcpy(&(beacon_packet.output_states), &(all_hk_data->EPS_hk.outputStatus), sizeof(uint16_t));
+  for (i = 0; i < 10; i++) {
+    memcpy(&(beacon_packet.output_faults[i]), &(all_hk_data->EPS_hk.outputFaultCnt[i]), sizeof(uint8_t));

```

Comment:
why use a loop instead of one memcpy?

Reply:
you're right, I was following what was done before. I'll change all the for loops to one memcpy